### ISSUE TITLE
clang-format inserts line break between constructor name and noexcept

### ISSUE THREAD

**Issue by beinhaerter** [llvm/llvm-project#55837]
clang-format 12, 13.0.1 and 14 format structs depending on their struct name.
`Rect` is formatted as expected/desired, but for `RECT` and `R` the `noexcept` of the constructor is put on the next line. It is not clear why `RECT` and `R` are handled differently than `Rect`.

Here is my C++ code.
```cpp
struct Rect {
	Rect() noexcept;
};

struct RECT {
	RECT()
	noexcept;
};

struct R {
	R()
	noexcept;
};
```

And here is the content of my `with .clang-format`.
```yaml
AccessModifierOffset: -4
AlignConsecutiveAssignments: AcrossComments
AlignConsecutiveBitFields: AcrossEmptyLinesAndComments
AlignConsecutiveDeclarations: false
AlignConsecutiveMacros: true
AlignOperands: true
AllowShortBlocksOnASingleLine: true
AllowShortCaseLabelsOnASingleLine: true
AllowShortFunctionsOnASingleLine: Inline
AllowShortIfStatementsOnASingleLine: false
AlwaysBreakTemplateDeclarations: true
BraceWrapping:
  AfterExternBlock: true
  AfterFunction: true
  BeforeCatch: true
  BeforeElse: true
  SplitEmptyFunction: false
  SplitEmptyRecord: false
  SplitEmptyNamespace: false
BreakBeforeBinaryOperators: NonAssignment
BreakBeforeBraces: Custom
BreakBeforeInheritanceComma: true
BreakConstructorInitializers: BeforeComma
BreakInheritanceList: BeforeComma
ColumnLimit: 0
Cpp11BracedListStyle: false
IndentGotoLabels: false
IndentPPDirectives: BeforeHash
IndentWidth: 4
KeepEmptyLinesAtTheStartOfBlocks: false
MaxEmptyLinesToKeep: 5
NamespaceIndentation: All
PointerAlignment: Left
ReflowComments: false
SpaceBeforeSquareBrackets: false
SpacesBeforeTrailingComments: 3
SpacesInConditionalStatement: false
TabWidth: 4
UseTab: true
```

**Comment by mydeveloperday** [comment]
This might be something to do with thinking R and RECT are potentially macros

**Comment by mydeveloperday** [comment]
So actually its the ColumnLimit: 0 thats causing this a clang-format file of just

`ColumnLimit: 0`

will show this..

```
struct Rect {
  Rect() noexcept;
};

struct RECT {
  RECT() noexcept;
};

struct R {
  R() noexcept;
};
```

vs

```
struct Rect {
  Rect() noexcept;
};

struct RECT {
  RECT()
  noexcept;
};

struct R {
  R()
  noexcept;
};
```

**Comment by mydeveloperday** [comment]
Looking deeper into the debug output the is a "CanBreak" between the `)` and no `except` but I'm not quite sure what is forcing that break seeing as you have a column limit of 0

```
AnnotatedTokens(L=0):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=struct L=6 PPK=2 FakeLParens= FakeRParens=0 II=0x199f39b39f0 Text='struct'
 M=0 C=1 T=Unknown S=1 F=0 B=0 BK=0 P=5020 Name=identifier L=11 PPK=2 FakeLParens= FakeRParens=0 II=0x199f39edf20 Text='Rect'
 M=0 C=0 T=StructLBrace S=1 F=0 B=0 BK=1 P=23 Name=l_brace L=13 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text='{'
----
AnnotatedTokens(L=1):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=identifier L=4 PPK=2 FakeLParens= FakeRParens=0 II=0x199f39edf20 Text='Rect'
 M=0 C=0 T=Unknown S=0 F=0 B=0 BK=0 P=23 Name=l_paren L=5 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text='('
 M=0 C=0 T=Unknown S=0 F=0 B=0 BK=0 P=140 Name=r_paren L=6 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text=')'
 M=0 C=0 T=TrailingAnnotation S=1 F=0 B=0 BK=0 P=170 Name=noexcept L=15 PPK=2 FakeLParens= FakeRParens=0 II=0x199f39a74e8 Text='noexcept'
 M=0 C=0 T=Unknown S=0 F=0 B=0 BK=0 P=23 Name=semi L=16 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text=';'
----
AnnotatedTokens(L=0):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=r_brace L=1 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text='}'
 M=0 C=0 T=Unknown S=0 F=0 B=0 BK=0 P=23 Name=semi L=2 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text=';'
----
AnnotatedTokens(L=0):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=struct L=6 PPK=2 FakeLParens= FakeRParens=0 II=0x199f39b39f0 Text='struct'
 M=0 C=1 T=Unknown S=1 F=0 B=0 BK=0 P=5020 Name=identifier L=11 PPK=2 FakeLParens= FakeRParens=0 II=0x199f39edf50 Text='RECT'
 M=0 C=0 T=StructLBrace S=1 F=0 B=0 BK=1 P=23 Name=l_brace L=13 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text='{'
----
AnnotatedTokens(L=1):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=identifier L=4 PPK=2 FakeLParens= FakeRParens=0 II=0x199f39edf50 Text='RECT'
 M=0 C=0 T=Unknown S=0 F=0 B=0 BK=0 P=23 Name=l_paren L=5 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text='('
 M=0 C=0 T=FunctionAnnotationRParen S=0 F=0 B=0 BK=0 P=140 Name=r_paren L=6 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text=')'
 M=0 C=1 T=TrailingAnnotation S=1 F=0 B=0 BK=0 P=170 Name=noexcept L=15 PPK=2 FakeLParens= FakeRParens=0 II=0x199f39a74e8 Text='noexcept'
 M=0 C=0 T=Unknown S=0 F=0 B=0 BK=0 P=23 Name=semi L=16 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text=';'
----
AnnotatedTokens(L=0):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=r_brace L=1 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text='}'
 M=0 C=0 T=Unknown S=0 F=0 B=0 BK=0 P=23 Name=semi L=2 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text=';'
----
AnnotatedTokens(L=0):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=struct L=6 PPK=2 FakeLParens= FakeRParens=0 II=0x199f39b39f0 Text='struct'
 M=0 C=1 T=Unknown S=1 F=0 B=0 BK=0 P=5020 Name=identifier L=8 PPK=2 FakeLParens= FakeRParens=0 II=0x199f39edf80 Text='R'
 M=0 C=0 T=StructLBrace S=1 F=0 B=0 BK=1 P=23 Name=l_brace L=10 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text='{'
----
AnnotatedTokens(L=1):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=identifier L=1 PPK=2 FakeLParens= FakeRParens=0 II=0x199f39edf80 Text='R'
 M=0 C=0 T=Unknown S=0 F=0 B=0 BK=0 P=23 Name=l_paren L=2 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text='('
 M=0 C=0 T=FunctionAnnotationRParen S=0 F=0 B=0 BK=0 P=140 Name=r_paren L=3 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text=')'
 M=0 C=1 T=TrailingAnnotation S=1 F=0 B=0 BK=0 P=170 Name=noexcept L=12 PPK=2 FakeLParens= FakeRParens=0 II=0x199f39a74e8 Text='noexcept'
 M=0 C=0 T=Unknown S=0 F=0 B=0 BK=0 P=23 Name=semi L=13 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text=';'
----
AnnotatedTokens(L=0):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=r_brace L=1 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text='}'
 M=0 C=0 T=Unknown S=0 F=0 B=0 BK=0 P=23 Name=semi L=2 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text=';'
```

**Comment by rymiel** [comment]
Fixed by f54d42abcf82e122ed0ac4688f01c1f4da5599f2
